File: app/src/test/java/com/fieldservice/app/presentation/tickets/TicketFilterTest.java
package com.fieldservice.app.presentation.tickets;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import com.fieldservice.app.domain.model.TicketStatus;

import org.junit.Test;

public class TicketFilterTest {

    @Test
    public void allMatchesEveryStatus() {
        for (TicketStatus status : TicketStatus.values()) {
            assertTrue(TicketFilter.ALL.matches(status));
        }
    }

    @Test
    public void pendingMatchesOnlyOpenLikeStatuses() {
        assertTrue(TicketFilter.PENDING.matches(TicketStatus.ASSIGNED));
        assertFalse(TicketFilter.PENDING.matches(TicketStatus.COMPLETED));
    }

    @Test
    public void completedMatchesFinishedStatusesOnly() {
        assertTrue(TicketFilter.COMPLETED.matches(TicketStatus.COMPLETED));
        assertTrue(TicketFilter.COMPLETED.matches(TicketStatus.CANCELLED));
        assertFalse(TicketFilter.COMPLETED.matches(TicketStatus.IN_PROGRESS));
    }
}

File: legacy-kotlin/app/src/test/java/com/fieldservice/app/presentation/tickets/TicketFilterTest.kt
package com.fieldservice.app.presentation.tickets

import com.fieldservice.app.domain.model.TicketStatus
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class TicketFilterTest {

    @Test
    fun `ALL matches every status`() {
        TicketStatus.entries.forEach { status ->
            assertTrue(TicketFilter.ALL.matches(status))
        }
    }

    @Test
    fun `PENDING matches only open-like statuses`() {
        assertTrue(TicketFilter.PENDING.matches(TicketStatus.ASSIGNED))
        assertFalse(TicketFilter.PENDING.matches(TicketStatus.COMPLETED))
    }

    @Test
    fun `COMPLETED matches finished statuses only`() {
        assertTrue(TicketFilter.COMPLETED.matches(TicketStatus.COMPLETED))
        assertTrue(TicketFilter.COMPLETED.matches(TicketStatus.CANCELLED))
        assertFalse(TicketFilter.COMPLETED.matches(TicketStatus.IN_PROGRESS))
    }
}
